src/metatensor/models/cli/eval.py

```diff
@@ -178,6 +178,10 @@ def eval_model(
         )
     logger.info("Setting up evaluation set.")
 
+    # TODO: once https://github.com/lab-cosmo/metatensor/pull/551 is merged and released
+    # use capabilities instead of this workaround
+    dtype = next(model.parameters()).dtype
```

Comment:
that's not how it will work in any of the other engines. The current idea is that the model decide which dtype it wants, and all inference will happen with this dtype.

For metatensor-models, this means that inference will happen with the same dtype used for training, and/or there can be a function to change the model dtype during export.

Reply:
Ok